### DISCUSSION TITLE
Support for Virtual List in Menu Component

### DISCUSSION THREAD

**Discussion by rodrigoea** [chakra-ui/chakra-ui#7852] [Ideas]
### Description

The current Menu component in Chakra-UI does not support virtual list rendering, resulting in performance issues when dealing with large lists. This issue aims to request the addition of virtual list support to enhance the performance of the Menu component.

### Problem Statement/Justification

Adding virtual list support would significantly improve the performance and user experience in such scenarios.

### Proposed Solution or API

I propose implementing a virtualized rendering mechanism in the Menu component 😄 

### Alternatives

I attempted to implement virtual list functionality by manually introducing code within the MenuOptionGroup component. However, this approach doesn't fully solve the problem and it resulted in scrolling issues.

Integrating virtual list support directly into the Menu component would provide a more reliable solution.

**Comment by baptooo** [comment]
Hello, I proposed a virtualized mechanism over `MenuOptionGroup` as I thought it was more appropriate rather than entire `Menu`
https://github.com/chakra-ui/chakra-ui/pull/8087
